tests/specializations/models/test_rdf_directory_service.py

```diff
@@ -0,0 +1,60 @@
+#
+# Blue Brain Nexus Forge is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Blue Brain Nexus Forge is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Blue Brain Nexus Forge. If not, see <https://choosealicense.com/licenses/lgpl-3.0/>.
+import os
+from pathlib import Path
+import rdflib
+from kgforge.specializations.models import RdfModel
+from kgforge.specializations.models.rdf.directory_service import (
+    _load_rdf_files_as_graph,
+)
+from tests.specializations.models.data import TYPES_SHAPES_MAP
+
+
+def test_load_rdf_files_as_graph(shacl_schemas_file_path):
+    graph_dataset = _load_rdf_files_as_graph(Path(shacl_schemas_file_path))
+    assert isinstance(graph_dataset, rdflib.Dataset)
+    shape_graphs = [str(g.identifier) for g in graph_dataset.graphs()]
+    assert len(shape_graphs) == 4
+    expected_file_paths = [
+        str(f.resolve())
+        for f in Path(shacl_schemas_file_path).rglob(os.path.join("*.*"))
+    ]
+    expected_named_graphs = expected_file_paths + ["urn:x-rdflib:default"]
+    assert sorted(shape_graphs) == sorted(expected_named_graphs)
+    for file_path in expected_file_paths:
+        g = graph_dataset.graph(file_path)
+        expected_g = rdflib.Graph().parse(file_path, format="json-ld")
+        assert len(g) == len(expected_g)
+
+
+def test_build_shapes_map(rdf_model_from_dir: RdfModel):
+    (class_to_shape, shape_to_defining_resource, defining_resource_to_named_graph) = (
+        rdf_model_from_dir.service._build_shapes_map()
+    )
+
+    assert sorted(list(class_to_shape.values())) == sorted(
```

Comment:
I guess if we accept that some shapes will target no class (which is the case for GeoShape), this assertion will not be True

Reply:
Yes. The test is adapted now